services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -598,21 +601,19 @@ getCommitData ::
     Member TinyLog r
   ) =>
   Local Data.Conversation ->
+  ConversationMLSData ->
   Epoch ->
   Commit ->
-  Sem r (GroupId, ProposalAction)
-getCommitData lconv epoch commit = do
-  convMeta <-
-    preview (to convProtocol . _ProtocolMLS) (tUnqualified lconv)
-      & noteS @'ConvNotFound
-
-  let curEpoch = cnvmlsEpoch convMeta
-      groupId = cnvmlsGroupId convMeta
+  Sem r ProposalAction
+getCommitData lconv mlsMeta epoch commit = do
+  let curEpoch = cnvmlsEpoch mlsMeta
+      groupId = cnvmlsGroupId mlsMeta
+      suite = cnvmlsCipherSuite mlsMeta
```

Comment:
This now allows to drop the `Member (ErrorS 'ConvNotFound) r` constraint from this function: it is not thrown directly here anymore, nor `applyProposalRef` needs to throw it because it does have all the MLS metadata needed.

Reply:
I could remove the `ConvNotFound` constraint from `checkEpoch`, but `checkGroup` still throws it. I refactored both functions to not used the converation object anymore but rather the `ConversationMLSData` in
https://github.com/wireapp/wire-server/pull/2859/commits/8e9278987a2a127391a9dbf952cdeaf9a1d04d48